### PR TITLE
feat: Elvish Shell support

### PR DESCRIPTION
**Pull request by elijahr**
# Summary

Add support for [Elvish Shell](https://elv.sh/).

### CODE REVIEW

**Comment by elijahr** [comment]
@jthegedus I've reworked this considerably, updated documentation, improved test coverage, and added shell completions. This is ready for review.

**Comment by jthegedus** [comment]
@elijahr I would like another maintainer to review this before merge. Unfortunately we're all rather busy all the time. Thanks for your patience.

File: .github/workflows/tests.yml
Comment (Stratus3D) [inline .github/workflows/tests.yml:44]: Is this building elvish from source? Or is this just downloading a pre-built binary?

File: asdf.fish
Comment (Stratus3D) [inline asdf.fish:5]: The first argument to `printf` should be the format string. The format string should not contain variables, only placeholders. For example, in this case `printf "%s\n" "$ASDF_DATA_DIR/shims"` to replace `echo`.

Comment (elijahr) [reply asdf.fish]: Changed in e697aaba1d15ea53efbddb31b382a7df0de379bb

File: .github/workflows/tests.yml
Comment (elijahr) [reply .github/workflows/tests.yml]: Building from source - didn't take too long locally, but looks like there is a linux binary available, so using that instead. Changed in 25382798d9b2d65a3867f548732a5f625f195d85

**Comment by jthegedus** [comment]
Going to merge this as it is a large change and the longer we leave it the more annoying it will become to merge. We can hotfix `head` with any major issues. Thanks @elijahr

**Review by Stratus3D — APPROVED** [review]

**Comment by Stratus3D** [comment]
Thanks @jthegedus and @elijahr ! Looks great!

**Pull request merged**
